### PR TITLE
Added a InfluxDBReporter

### PR DESCRIPTION
**Pull request by liyinan926**
Note that the deleted `commonsHttpClient` is a duplicated entry.

Signed-off-by: Yinan Li <liyinan926@gmail.com>

### CODE REVIEW

**Comment by liyinan926** [comment]
@ibuenros, can you take a look at this pull request?

File: gobblin-metrics/src/main/java/gobblin/metrics/influxdb/InfluxDBReporter.java [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+  private Serie.Builder addMetered(String name, Metered metered, long timeStamp, Serie.Builder builder) {
+    return builder
+        .values(timeStamp, MetricRegistry.name(name, Measurements.COUNT.getName()), metered.getCount())
+        .values(timeStamp, MetricRegistry.name(name, Measurements.RATE_1MIN.getName()), metered.getOneMinuteRate())
+        .values(timeStamp, MetricRegistry.name(name, Measurements.RATE_5MIN.getName()), metered.getFiveMinuteRate())
+        .values(timeStamp, MetricRegistry.name(name, Measurements.RATE_15MIN.getName()), metered.getFifteenMinuteRate())
+        .values(timeStamp, MetricRegistry.name(name, Measurements.MEAN_RATE.getName()), metered.getMeanRate());
+  }
+
+  private Serie.Builder reportSnapshot(String name, Snapshot snapshot, long timeStamp, Serie.Builder builder) {
```
Comment (ibuenros) [inline gobblin-metrics/src/main/java/gobblin/metrics/influxdb/InfluxDBReporter.java]: can we change the method name to addSnapshot for consistency?

**Comment by ibuenros** [comment]
Does InfluxDB not support tags?

**Comment by liyinan926** [comment]
Currently no, but the next version will support tags.

File: gobblin-metrics/src/main/java/gobblin/metrics/influxdb/InfluxDBReporter.java (hunk 1 above)
Comment (liyinan926) [inline gobblin-metrics/src/main/java/gobblin/metrics/influxdb/InfluxDBReporter.java]: Fixed.

**Comment by liyinan926** [comment]
Updated.

**Comment by ibuenros** [comment]
Given the lack of tags, would it make sense to add the tags to the names of the metrics?

**Comment by liyinan926** [comment]
The tags will be included if the `MetricContext` the reporter is associated to is configure to `reportFullyQualifiedNames`.

**Comment by ibuenros** [comment]
Ok, this looks good to me then.

**Pull request merged**